added feature to systemd's journalctl to alter the logline

GalaxyGorilla opened the pull request:
See commit message in patch for more information.

GalaxyGorilla committed:
added feature to systemd's journalctl to alter the appearance of the logline

thz commented:
cool thing. did that patch land upstream already? can you reference commit or PR?

GalaxyGorilla commented:
The patch was already in discussion on the mailing list, see

http://lists.freedesktop.org/archives/systemd-devel/2014-September/023327.html

I just aligned it with systemd 219 and did some minor fixes. I also changed the query language a little bit. It was not accepted because of some flaws in the language and the used type system. I fixed those parts but the folks from systemd want the formatting to be reusable for things like the message catalog.

GalaxyGorilla commented:
Just tested with the meta-caros layer. Seems to work without flaws.

thz commented:
upstreaming this sounds like a perfect assignment for the upcoming hackathon

GalaxyGorilla commented on recipes-core/systemd/systemd/logline-formatting.patch:
```diff
@@ -0,0 +1,1095 @@
++                                        type = OUTPUT_FIELD_RAW;
++                        }
++
++                        if ((r = output_formatter_add_field(formatter, type, vallen, data, datalen)) < 0) {
```
'vallen' is not initialized here when the above 'else' branch is taken.

GalaxyGorilla committed:
fixed uninitialized variable
